controllers/osbuildconfig_controller.go

```diff
@@ -125,11 +147,134 @@ func (r *OSBuildConfigReconciler) createNewOSBuildCR(ctx context.Context, osBuil
 		return err
 	}
 
+	if kickstartConfigMap != nil {
+		err = r.setKickstartConfigMapOwner(ctx, kickstartConfigMap, osBuild)
+		if err != nil {
+			logger.Error(err, "cannot set controller reference to kickstart config map")
+			return err
+		}
+	}
+
 	logger.Info("A new OSBuild CR was created", "OSBuild", osBuild.Name)
 
 	return nil
 }
 
+func (r *OSBuildConfigReconciler) applyTemplate(ctx context.Context, osBuildConfig *osbuilderprojectflottaiov1alpha1.OSBuildConfig, osBuildConfigSpecDetails *osbuilderprojectflottaiov1alpha1.BuildDetails, osBuildName string, osBuild *osbuilderprojectflottaiov1alpha1.OSBuild) (*v1.ConfigMap, *osbuilderprojectflottaiov1alpha1.OSBuildConfigTemplate, error) {
+	var kickstartConfigMap *v1.ConfigMap
+	var osConfigTemplate *osbuilderprojectflottaiov1alpha1.OSBuildConfigTemplate
+	buildConfigCustomizations := osBuildConfigSpecDetails.Customizations
+	if template := osBuildConfig.Spec.Template; template != nil {
+		var err error
+		osConfigTemplate, err = r.OSBuildConfigTemplateRepository.Read(ctx, template.OSBuildConfigTemplateRef, osBuildConfig.Namespace)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		mergeCustomization(osConfigTemplate, osBuildConfigSpecDetails, buildConfigCustomizations)
+
+		kickstartConfigMap, err = r.createKickstartConfigMap(ctx, osBuildConfig, osConfigTemplate, osBuildName, osBuild.Namespace)
+		if err != nil {
+			return nil, nil, err
+		}
+		if kickstartConfigMap != nil {
+			osBuild.Spec.Kickstart = &osbuilderprojectflottaiov1alpha1.NameRef{Name: osBuildName}
```

Comment:
Maybe the name needs to be kickstartConfigMap's name? (here in your case it's "osBuildName" )